src/Microsoft.Health.Dicom.Api/Features/Routing/UrlResolver.cs

```diff
@@ -46,9 +46,10 @@ public sealed class UrlResolver : IUrlResolver
         public Uri ResolveRetrieveStudyUri(string studyInstanceUid)
         {
             EnsureArg.IsNotNull(studyInstanceUid, nameof(studyInstanceUid));
+            var hasVersion = _httpContextAccessor.HttpContext.Request.RouteValues.ContainsKey("version");
 
             return RouteUri(
-                KnownRouteNames.RetrieveStudy,
+                hasVersion ? KnownRouteNames.VersionedRetrieveStudy : KnownRouteNames.RetrieveStudy,
                 new RouteValueDictionary()
                 {
                     { KnownActionParameterNames.StudyInstanceUid, studyInstanceUid },
```

Comment:
We would want to propagate the version if it was passed, right? Here and below

Reply:
By getting the route name, it will be able to grab the version used. Just double checked that it works with with multiple versions registered (at least, it works in the E2E tests)

Reply:
Oh, would `UrlHelper.RouteUrl(...)` return a URL with version we recevied already populated if you pass `KnownRouteNames.VersionedRetrieveStudy` as the name? I wonder how it would do that lol. I assume it would grab the corresponding template `"v{version:apiVersion}/study/{studyInstaneUid}"`, then we pass it a mapping of route values to resolve the template.

If that works, does it work for non-default versions?

Reply:
Yeah I tested non-default versions by having [ApiVersion("1.0")] and [ApiVersion("1.0-prerelease")] on the StoreController (with prerelease being the default). It seems to just magically known which version was used without anything needing to be passed in. 
The URL returned in the postman response is:
 - `baseUrl/studies/...` for the non-versioned route (KnownRouteNames.RetrieveStudy)
 - `baseUrl/v1.0-prerelease/studies/...` for 1.0-prerelease
 - `baseUrl/v1.0/studies/...` for 1.0

I've updated the tests in `StoreTransactionTests` that call `ValidateReferencedSopSequence` to ensure that the URLs match their version